pdpserver/server/service.go

```diff
@@ -180,13 +171,14 @@ func (s *Server) Validate(ctx context.Context, in *pb.Msg) (*pb.Msg, error) {
 	s.RUnlock()
 
 	if s.opts.autoResponseSize {
-		msg.Body = s.rawValidate(p, c, in.Body)
+		msg.Body, err = s.rawValidate(p, c, in.Body)
 		return msg, err
 	}
 
 	b := s.pool.Get()
-	msg.Body = s.rawValidateToBuffer(p, c, in.Body, b)
+	body, err := s.rawValidateToBuffer(p, c, in.Body, b)
```

Comment:
Here you define new `err` variable, which shadows existing one. Can you make sure that this doesn't break existing logic, e.g. `validatePostHook` logic?

Reply:
fixed